test/irb/test_context.rb

```diff
@@ -668,7 +668,16 @@ def test_build_completor
       assert_equal 'IRB::RegexpCompletor', @context.send(:build_completor).class.name
       IRB.conf[:COMPLETOR] = :type
       if RUBY_VERSION >= '3.0.0' && RUBY_ENGINE != 'truffleruby'
-        assert_equal 'IRB::TypeCompletion::Completor', @context.send(:build_completor).class.name
+        begin
+          require 'repl_type_completor'
+          type_completor_available = true
+        rescue LoadError
+        end
+        if type_completor_available
+          # Load RBS before testing to avoid completor preload RBS in a thread
+          ReplTypeCompletor.load_rbs unless ReplTypeCompletor.rbs_loaded?
+          assert_equal 'IRB::TypeCompletor', @context.send(:build_completor).class.name
+        end
```

Comment:
I feel we can move this to `test_type_completor.rb` and make it an integration test. This way we won't need the condition.

Reply:
Sounds nice, moved.
Changed to test the `irb_info` result and also test if `@completor.completion_candidates` is really working